manager/usermanager/updateUser.go

```diff
@@ -12,9 +12,27 @@ import (
 
 // UpdateUserDetails get the user information
 func UpdateUserDetails(userStore *store.UserStore, user *models.UserCredentials) (*models.PublicUserInfo, error) {
-	// This is just for github auth. Will implement a switch  case here once we have a self signup
-	if user.OnBoardingState == models.BoardingStateEmailVerified && user.Kind == models.GithubAuth {
-		user.OnBoardingState = models.BoardingStateVerifiedAndComplete
+	switch user.OnBoardingState {
+	case models.BoardingStateSignup:
+		{
+			if user.Email != nil {
+				user.OnBoardingState = models.BoardingStateEmailVerified
```

Comment:
You are changing the onboarding state within the switch clause, is that fine?

Reply:
I guess it should be fine as its a pointer

Reply:
Will execution hit the relevant switch case when the change happens within switch block?
Eg, will execution go from line no. 22 to line. 28?
Or from line 24 to line 35?

Reply:
No @avishnu Only one of the switch case gets executed at a time in golang. It works as if breaak statement is hidden.